app/models/ticket.rb

```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+
+class Ticket < ApplicationRecord
+  enum :ticket_type, {
+    edit_person: "edit_person",
+  }
+
+  has_many :ticket_logs
+  has_many :ticket_stakeholders
+  belongs_to :metadata, polymorphic: true
+
+  def user_stakeholders(user)
+    return [] if user.nil?
+    ticket_stakeholders.select do |stakeholder|
+      if stakeholder.stakeholder_type == :user
+        stakeholder.stakeholder_id == user.id
+      elsif stakeholder.stakeholder_type == TicketStakeholder.stakeholder_types[:user_group]
+        user.active_roles.any? { |role| role.group_id == stakeholder.stakeholder_id }
+      end
+    end
+  end
+
+  def action_allowed?(action, user)
+    user_stakeholders(user).any? do |stakeholder|
+      (
+        stakeholder.stakeholder_type == TicketStakeholder.stakeholder_types[:user_group] &&
+        metadata.action_user_groups(action).include?(stakeholder.stakeholder_id)
+      )
+    end
+  end
+
+  def url
+    Rails.application.routes.url_helpers.ticket_url(id)
+  end
```

Comment:
Hm, I'm not a big fan of "forcing" routes into AR models.

There's a reason that you have to call it in such a complicated way through `Rails.application.routes.url_helpers`, and that reason is: Helpers aren't supposed to be available in AR models.

Most places in our code that need this URL (like views, email templates, etc.) have these helpers available and mounted without needing the `Rails.application.routes.url_helpers` hack, so maybe just use `ticket_url` directly there, instead of "sideloading" it here.